sdio/imx6ull/imx6ull-sdio.c

```diff
@@ -0,0 +1,855 @@
+/*
+ * Phoenix-RTOS
+ *
+ * i.MX 6ULL SDIO driver
+ *
+ * Copyright 2022 Phoenix Systems
+ * Author: Ziemowit Leszczynski, Artur Miller
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+#include <stdio.h>
+#include <string.h>
+#include <unistd.h>
+#include <errno.h>
+
+#include <sys/mman.h>
+#include <sys/platform.h>
+#include <sys/threads.h>
+#include <sys/interrupt.h>
+
+#include <phoenix/arch/imx6ull.h>
+
+#include <sdio.h>
+
+
+/* SDIO standard commands */
+#define SDIO_CMD_STATE_IDLE     0  /* set the card into idle state       */
+#define SDIO_CMD_STATE_INACTIVE 15 /* set the card into inactive state   */
+#define SDIO_CMD_SELECT_CARD    7  /* select a card by address           */
+#define SDIO_CMD_RELATIVE_ADDR  3  /* set card's relative address        */
+#define SDIO_CMD_SEND_OP_COND   5  /* inquire about operating conditions */
+#define SDIO_CMD_RW_DIRECT      52 /* direct register I/O operation      */
+#define SDIO_CMD_RW_EXTENDED    53 /* bulk register I/O operation        */
+
+#define DMA_BUFFER_SIZE     2048
+#define THREAD_STACK_SIZE   1024
+#define SDIO_IRQ_STS_CARD   (1 << eventBits[SDIO_EVENT_CARD_IRQ])
+#define SDHC_RETRIES        10
+#define ARRAY_LENGTH(array) (sizeof(array) / sizeof(array[0]))
+
+/* hardware platform specific */
+#define USDHC2_ADDR             0x2194000
+#define USDHC2_IRQ              (32 + 23)
+#define SDHC_SYSCTL_RESET_ALL   (1 << 24)
+#define SDHC_SYSCTL_RESET_CMD   (1 << 25)
+#define SDHC_SYSCTL_RESET_DATA  (1 << 26)
+#define SDHC_SYSCTL_CLK_DIV_8   (0x04 << 8)
+#define SDHC_SYSCTL_CLK_DIV_4   (0x02 << 8)
+#define SDHC_SYSCTL_CLK_DEFAULT 0x8010
+#define SDHC_SYSCTL_RESERVED    0xf
+#define SDHC_INTERRUPT_DEFAULTS 0x107f000f
+#define SDHC_CMD_ERROR          0x107f0000
+
+/* maps interrupt event to bit number in registers */
+static const uint8_t eventBits[19] = { 6, 7, 8, 1, 4, 5, 0, 16, 17, 18, 19, 20, 21, 22, 3, 28, 12, 26, 2 };
```

Comment:
You don't have to put an array size here:
```suggestion
static const uint8_t eventBits[] = { 6, 7, 8, 1, 4, 5, 0, 16, 17, 18, 19, 20, 21, 22, 3, 28, 12, 26, 2 };
```

Reply:
Removed.